MODAK/src/jobfile_generator.py

```diff
@@ -3,6 +3,75 @@
 from tuner import tuner
 import logging
 
+"""Constant to represent the scheduler to use is SLURM"""
+SCHEDULER_SLURM="slurm"
+"""Constant to represent the scheduler to use is PBS Torque"""
+SCHEDULER_TORQUE="torque"
+
+class ArgumentConverter():
+        # Define which options are vaild of each of the schedulers
+    VAILD_TORQUE=['', 'a','b','e'] # abort, begin, end
+    VALID_SLURM=["", "NONE", "BEGIN", "END", "FAIL", "REQUEUE", "ALL",
+        "INVALID_DEPEND", "STAGE_OUT", "TIME_LIMIT", "TIME_LIMIT_90", 
+        "TIME_LIMIT_80", "TIME_LIMIT_50", "ARRAY_TASKS"]
+    SLURM_TO_TORQUE={
+        "NONE": "",
+        "BEGIN": "b", 
+        "END": "e", 
+        "FAIL": "a",
+        "REQUEUE": "",
+        "ALL": "abe",
+        "INVALID_DEPEND": "a",
+        "STAGE_OUT": "",
+        "TIME_LIMIT": "a",
+        "TIME_LIMIT_90": "", 
+        "TIME_LIMIT_80": "",
+        "TIME_LIMIT_50": "",
+        "ARRAY_TASKS": ""
+    }
+    TORQUE_TO_SLURM={
+        "a": "FAIL,INVALID_DEPEND,TIME_LIMIT",
+        "b": "BEGIN",
+        "e": "END"
+    }
+
+    def _slurm_to_torque(self, options):
+        # For each option in the slurm options, look it up in the 
+        # SLURM_TO_TORQUE dict. Then concatenate all the returned values
+        return "".join([self.SLURM_TO_TORQUE[opt] for opt in options.split(",")])
+
+    def _torque_to_slurm(self, options):
+        # For each character in the torque options, look it up in the
+        # TORQUE_TO_SLURM dict. Then concatenate all the returned vaules,
+        # separating them with commas 
+        return ",".join([self.TORQUE_TO_SLURM[opt] for opt in options])
+
+    def _is_slurm_options(self, options):
+        # Split the options string on commas, then check each part is a valid
+        # option. Return true if all are; false otherwise.
+        return all(opt in self.VALID_SLURM for opt in options.split(","))
+    
+    def _is_torque_options(self, options):
+        # For each character in the options, check if it is a valid torque
+        # option. If all are, return true, otherwise, false
+        return all(opt in self.VAILD_TORQUE for opt in options)
+
+    def convert_notifications(self, scheduler, notifications):    
+        # Check if the options we have match the scheduler we have
+        if scheduler==SCHEDULER_SLURM and self._is_torque_options(notifications) and notifications!="":
+            # Scheduler is slurm, but notifications are torque.
+            # convert then return
+            return self._torque_to_slurm(notifications)
+        elif scheduler==SCHEDULER_TORQUE and self._is_slurm_options(notifications) and notifications!="":
+            # Scheduler is torque, notifications are slurm.
+            # convert, then return
+            return self._slurm_to_torque(notifications)
+        # Scheduler and options match; return the original notification options
+        # (Also, if notifications are invalid this leaves them intact/unmodified)
```

Comment:
If notifications are invalid, the job script will also be invalid and won't run. In this case, I think we should not include notifications in the job script if it is invalid.

Reply:
Would you like me to filter them if they're invalid? This code currently allows invalid notifications to pass so that it doesn't change how they're handled. If I change this to filter invalid notifications, it will change the behavior of MODAK as a whole.